testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -151,6 +151,18 @@ def test_toggle_atom_nomatch_mismatch_atoms(self, universe, reference):
         with pytest.raises(SelectionError):
             align.alignto(u, ref, select='all', match_atoms=False)
 
+    @pytest.mark.parametrize('sub_string',('resname ALA and name CA', 1234,
+    mda.Universe(PSF, DCD).select_atoms('resname ALA and name CA')))
+    def test_subselection_alignto(self, universe, reference, sub_string):
+        
+        if isinstance(sub_string, int):
+            with pytest.raises(TypeError):
```

Comment:
for conditional parametrized raising see here: http://doc.pytest.org/en/latest/example/parametrize.html#parametrizing-conditional-raising